mne/viz/ica.py

```diff
@@ -130,6 +130,223 @@ def plot_ica_sources(ica, inst, picks=None, exclude=None, start=None,
+    power spectrum and epoch variance.
+
+    Parameters
+    ----------
+    inst: instance of Epochs or Raw
+        The data to use in plotting properties.
+    ica : instance of mne.preprocessing.ICA
+        The ICA solution.
+    picks : int | array-like of int | None.
+        The components to be displayed. If None, plot will show the
+        sources in the order as fitted. If more than one components were
+        chosen in the picks - each one will be plotted in a separate figure.
+        Defaults to None.
+    axes: list of matplotlib axes | None
+        List of five matplotlib axes to use in plotting: [topo_axis,
+        image_axis, erp_axis, spectrum_axis, variance_axis]. If None a new
+        figure with relevant axes is created. Defaults to None.
+    dB: bool
+        Whether to plot spectrum in dB. Defaults to False.
+    cmap: matplotlib colormap | None
+        Colormap to use in both topoplot and epochs image. If None topoplot
+        and epochs image use their default "RdBu_r" colormap. Defaults to None.
+    plot_std: bool | float
+        Whether to plot standard deviation in ERP/ERF and spectrum plots.
+        Defaults to True, which plots one standard deviation above/below.
+        If set to float allows to control how many standard deviations are
+        plotted. For example 2.5 will plot 2.5 standard deviation above/below.
+    topo_kws : dict | None
+        Dictionary of arguments to plot_topomap. If None - doesn't pass any
+        additional arguments. Defaults to None.
+    image_kws : dict | None
+        Dictionary of arguments to plot_epochs_image. If None - doesn't pass
+        any additional arguments. Defaults to None.
+    show : bool
+        Show figure if True.
+
+    Returns
+    -------
+    fig : instance of pyplot.Figure
+        The figure.
+
+    """
+    from .epochs import plot_epochs_image
+    from ..io.base import _BaseRaw
+    from ..epochs import _BaseEpochs
+    from ..preprocessing import ICA
+    from ..time_frequency.psd import psd_multitaper
+
+    # current component/channel (with future interactivity in mind)
+    idx = 0
+
+    if not isinstance(inst, (_BaseRaw, _BaseEpochs)):
+        raise ValueError('inst should be an instance of Raw or Epochs,'
+                         ' got %s instead.' % type(inst))
+    elif not isinstance(ica, ICA):
+        raise ValueError('ica has to be an instance of ICA, '
+                         'got %s instead' % type(ica))
+    else:
+        from .topomap import _plot_ica_topomap
+    if not isinstance(plot_std, (bool, float, int)):
+        raise ValueError('plot_std has to be a bool, int or float, '
+                         'got %s instead' % type(ica))
+    elif isinstance(plot_std, bool) and plot_std:
+        num_std = 1.
+    elif isinstance(plot_std, (float, int)):
+        num_std = plot_std
+        plot_std = True
+
+    picks = range(ica.n_components_) if picks is None else picks
```

Comment:
I can cast it to a list - but it should work with range, right? Both indexing a range and indexing with an array with range are valid operations.